### PR TITLE
Implement automatic command documentation

### PR DESCRIPTION
**Pull request by linuswillner**
* The command reference is now mostly generated based on the source code of the commands, namely their metadata. Some parts are still manually documented, will look into autogen for those later.
* Wrote a few additional utils and changed some things, namely:
    - Added src/internal/command-info-appender.js to take the data from commandInfo.json and forward it to the autogen script
    - Added docs/js/src/commands.js to autogen the HTML required for the command documentation
    - Made a modification in src/internal/info-dump.js that recasts the command info JSON file as commandInfo.json (More declarative naming)
* Added a script to minify the command docs autogen JS with Uglifyjs
* Added non-essential files created by docs autogen to .gitignore and .eslintignore
* Slightly modified CI config to run with 'machine' that provides access to both Node and Python 3.5 at the same time
* Some commands have received tweaks to their metadata to accurately document their functionality

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -80,8 +80,8 @@ jobs:
             . venv/bin/activate
             mkdocs build
   docs-deploy:
-    docker:
-      - image: python:3.5
+    machine:
```
Comment (Dougley) [inline .circleci/config.yml]: Did you test this? Why is this change needed?

File: docs/js/dist/.empty_directory [hunk 2]
```diff
@@ -0,0 +1 @@
+Dist versions of JavaScript files in the 'src' directory will go here. It's generally a bad idea to modify the files in this directory.
```
Comment (Dougley) [inline docs/js/dist/.empty_directory]: Name this file `.gitempty`, this is purely for style.

File: .circleci/config.yml (hunk 1 above)
Comment (linuswillner) [reply .circleci/config.yml]: This has not been tested with this particular setup because I haven't had the opportunity to use it before. However, I consulted relevant documentation and verified it has been executed in the manner Circle desires.

As for why this change is required, the `docs-deploy` script now requires Node in addition to Python. This is due to the fact that the command metadata and docgen scripts are exempt from version control because they're not relevant to the operation of the app. As such they are non-persistent and must be run prior to GitHub Pages deployment (Which removes the ability to run the autogen scripts in a separate service). If the Python Docker image is used, Node is not available, and vice versa. Specifying Node as a secondary container will not work either as per CircleCI, since all build scripts run in the primary container.

The `machine` system, on the other hand, has several language tools pre-installed, among those Node and Python 3.5. In this case I saw fit to use `machine` despite it having a slightly longer startup time of approximately 30-60 seconds, which I think our workflow can absorb quite well.
